apps/web/pages/settings/organizations/platform/oauth-clients/create.tsx

```diff
@@ -1,14 +1,14 @@
-import { useRouter } from "next/router";
 import React from "react";
 
 import { getLayout } from "@calcom/features/settings/layouts/SettingsLayout";
+import { useCompatSearchParams } from "@calcom/lib/hooks/useCompatSearchParams";
 
 import PageWrapper from "@components/PageWrapper";
 import { OAuthClientForm } from "@components/settings/organizations/platform/oauth-clients/OAuthClientForm";
 
 export const CreateOAuthClient = () => {
-  const router = useRouter();
-  const clientId = router.query.clientId as string | undefined;
+  const searchParams = useCompatSearchParams();
+  const clientId = searchParams?.get("clientId") || "";
```

Comment:
Question: when is the "clientId" set? I see below is OAuthClientForm and its create.ts file used to create oauth client, so when do we need clientId if we are creating a new client?

Reply:
clientId is set when clicking on the oauuth client card edit button, we currently reuse the same create form to update